src/PHPUnit/Traits/With_Post_Remapping.php

```diff
@@ -350,4 +350,44 @@ protected function get_mock_organizer( $target, array $template_vars = null ) {
 			? tribe_get_organizer_object( $remapped_id )
 			: get_post( $remapped_id );
 	}
+
+	/**
+	 * Inserts a post in the database with a direct query.
+	 *
+	 * This is useful to create real versions of mock posts.
+	 *
+	 * @since TBD
+	 *
+	 * @param \WP_Post $post The post object to insert.
+	 */
+	protected function wp_insert_post( \WP_Post $post ) {
+		global $wpdb;
+		$post_fields         = [
+			'ID',
+			'post_author',
+			'post_date',
+			'post_date_gmt',
+			'post_content',
+			'post_title',
+			'post_excerpt',
+			'post_status',
+			'comment_status',
+			'ping_status',
+			'post_password',
+			'post_name',
+			'to_ping',
+			'pinged',
+			'post_modified',
+			'post_modified_gmt',
+			'post_content_filtered',
+			'post_parent',
+			'guid',
+			'menu_order',
+			'post_type',
+			'post_mime_type',
+			'comment_count',
+		];
+		$array_intersect_key = array_intersect_key( (array) $post, array_combine( $post_fields, $post_fields ) );
+		$wpdb->insert( $wpdb->posts, $array_intersect_key );
```

Comment:
Any particular reason of using `$wpdb` instead of `wp_insert_post()`?

Reply:
Yes, the `wp_insert_post` function is a wrapper that will update cache; we are spoofing the cache and need it to be left untouched.